instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/CgroupV2ContainerIdExtractor.java

```diff
@@ -42,15 +42,26 @@ Optional<String> extractContainerId() {
       return empty();
     }
     try {
-      return filesystem
-          .lines(V2_CGROUP_PATH)
-          .filter(line -> line.contains("hostname"))
-          .flatMap(line -> Stream.of(line.split("/")))
-          .map(CONTAINER_ID_RE::matcher)
-          .filter(Matcher::matches)
+      List<String> fileAsList = filesystem.lineList(V2_CGROUP_PATH);
```

Comment:
I'd move the rest of the code out of the try catch block, something like
```java
List<String> fileAsList;
try {
  fileAsList = filesystem.lineList(V2_CGROUP_PATH);
} catch (IOException e) {
  logger.log(Level.WARNING, "Unable to read v2 cgroup path", e);
  return empty();
}
Optional<String> optCid =
...
```

Reply:
That's OK. Fixed.